### PR TITLE
[SHELL32] Add 'Properties' menu item in My Computer

### PR DESCRIPTION
**Pull request by katahiromz**
## Purpose
This implements the 'Properties...' menu item in My Computer.
JIRA issue: [CORE-12509](https://jira.reactos.org/browse/CORE-12509)

## Proposed changes

- Add the Shell_DefaultContextMenuCallBack function.
- Implement IContextMenuCB interface in CDrivesFolder.

### CODE REVIEW

**Comment by yagoulas** [comment]
It would be better to use the def context menu with and pass a callback to add the properties item.

**Comment by katahiromz** [comment]
I don't know how to.

**Comment by blackcrack** [comment]
looks good and it's a good improvement :+1:

**Comment by katahiromz** [comment]
Done.

File: dll/win32/shell32/CDefaultContextMenu.cpp [hunk 1]
```diff
@@ -1514,3 +1515,38 @@ CDefFolderMenu_Create2(
...
+{
+    PIDLIST_ABSOLUTE pidlFolder;
+    PUITEMID_CHILD *apidl;
+    UINT cidl;
+    HRESULT hr = SH_GetApidlFromDataObject(pdtobj, &pidlFolder, &apidl, &cidl);
+    if (FAILED_UNEXPECTEDLY(hr))
+        return hr;
+
+    if (cidl > 1)
+        ERR("SHMultiFileProperties is not yet implemented\n");
```
Comment (learn-more) [inline dll/win32/shell32/CDefaultContextMenu.cpp]: What happens if we return an error here?

**Comment by HBelusca** [comment]
@yagoulas : could you told us whether the new version here is implemented according to what you thought?

File: dll/win32/shell32/CDefaultContextMenu.cpp (hunk 1 above)
Comment (katahiromz) [reply dll/win32/shell32/CDefaultContextMenu.cpp]: Property sheet doesn't open if selection was multiple.
Memory leaks.

Comment (katahiromz) [reply dll/win32/shell32/CDefaultContextMenu.cpp]: OK. It will return E_FAIL.

**Comment by katahiromz** [comment]
Done.

**Comment by yagoulas** [comment]
Yes, that's what I asked for :D. If you could move Shell_DefaultContextMenuCallBack to shlfolder.cpp it would be ready to merge.

**Comment by katahiromz** [comment]
Ok.
■

**Pull request merged**